packages/product/src/repositories/product-image.ts

```diff
@@ -97,7 +97,7 @@ export class ProductImageRepository extends AbstractBaseRepository<Image> {
         newImgs.push(this.manager_.create(Image, img))
       })
 
-      await this.manager_.persist(newImgs)
+      await this.manager_.persistAndFlush(newImgs)
```

Comment:
Yes it was intended, the flush is done by the transaction wrapper allowing us to not reach the db unless we need that and therefore reduce the transaction lifetime. And thanks to the entity map etc during the creation we don't need to flush till the end

Reply:
Did you add the flush for a specific need at that moment? Theoretically, once they are found the images becomes managed by the entity manager, and the ones that are created as well, so i don't see the need to flush here right away and it can be delayed to before the commit as it was.

Reply:
Cc @carlos-r-l-rodrigues

Reply:
For the update case, upserts for relations were not being flushed. I might've screwed up the transaction hierarchy, I'll take a look once again.

Reply:
This is probably my fault, in the repositories instead of this.manager_ we should have const manager = context.transactionManager ?? This.manager_ and use that one, also passing the context to the subsequent method calls to this.*

Reply:
Can i get you to change that in The main pr

Reply:
sure, will do. 

Won't the create method have also failed for this?

Reply:
All good, I'll take care of this. Focus on the workflows. 🔥

Reply:
What's the conclusion? Do we need to flush in each method, or can we let the transaction wrapper take care of it?

Reply:
We don't need to flush, just play with the references. We huddled around it and it works without flushing. In some scenario we will have to flush, but so far it is not needed

Reply:
No, not needed. If we work with objects instead of ids, mikroorm takes care of managing the sequence correctly.

https://github.com/medusajs/medusa/commit/94c2b745f99170eefd66d60bd370becad89a13cb